scripts/components/map.js

```diff
@@ -64,18 +64,18 @@ var BostonMap = (function () {
     var zoom = mapObj.componentZoom ? mapObj.componentZoom : mapObj.esriZoom ? mapObj.esriZoom : 14;
 
     // Apply default coordinates and zoom level.
-    map[map[0].id] = L.map(mapID, {zoomControl: false}).setView([latitude, longitude], zoom);
+    map.mapID = L.map(mapID, {zoomControl: false}).setView([latitude, longitude], zoom);
```

Comment:
Gotta clean this up a bit more. `map[0].id` is used above still. Also, this is setting the value of a `mapID` property to a Leaflet Map object, which will be confusing down the road.

Reply:
LMK if you want to chat / debug in person.

Reply:
Thanks for the help on this @finneganh! I updated the variable names so it's more clear what they represent, which simplifies the conditional logic. 

I also only declared the map ID once even though this value can be found in the object that Drupal passes via `mapData` or in the page template.

Reply:
@jimafisk Don't forget to change these bits to just `var map = L.map(…)`

Reply:
Ah yes, I see how that's unnecessary now. Should be all set. Thanks for pointing that out!